### PR TITLE
remove bad pattern match

### PR DESCRIPTION
**Pull request by vmchale**
This allows compilation with GHC 8.6.1, among other things.

### CODE REVIEW

**Comment by michaelpj** [comment]
This also loses the benefit of sharing the boolean generation code, which would have prevented some earlier bugs. Can't we do better?

**Comment by vmchale** [comment]
We could rewrite `getBuiltinBool` to return a `Type TyNameWithKind ()` if we wanted.

**Comment by michaelpj** [comment]
I think the fact that I had commented on this suggests that it was not uncontroversial, so I don't think this was an appropriate self-merge

File: language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs [hunk 1]
```diff
@@ -39,34 +37,51 @@ isType Type{} = True
...
+    nam <- newTyName (Size ())
     let ity = TyApp () (TyBuiltin () TyInteger) (TyVar () nam)
         fty = TyFun () ity (TyFun () ity ity)
-    Right t <- runExceptT $ annotateType $ TyForall () nam (Size ()) fty
-    pure t
+    pure $ TyForall () nam (Size ()) fty
 
 -- | Create a new 'Type' for an integer relation
-intRel :: MonadQuote m  => m (Type TyNameWithKind ())
+intRel :: (MonadQuote m)  => m (Type TyNameWithKind ())
```
Comment (effectfully) [inline language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs]: Do we really need these additional `()`?

**Comment by effectfully** [comment]
Also right now some stdlib terms contain duplicate uniques, because I share names inside terms, so this version is correct unlike the one in the stdlib. I do not fix the stdlib, because once I finish all the current CEK machine business, I'll show the huge difference between what we have now and we really should have.

**Comment by michaelpj** [comment]
If the thing that this depends on is wrong then we should fix that, not randomly revert stuff.

(Off topic: Roman, I haven't noticed that, maybe you could send me an example separately?)

**Comment by vmchale** [comment]
My apologies. I have opened a ticket to patch the patch.

File: language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs (hunk 1 above)
Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs]: Oh, no. That is a mistake.

**Pull request merged**